client/state/data-layer/wpcom/signup/verticals/index.js

```diff
@@ -25,14 +25,16 @@ export const requestVerticals = action =>
 			path: '/verticals',
 			query: {
 				search: action.search.trim(),
+				site_type: action.siteType,
```

Comment:
I think this is okay for now, but it might change later depending on how we refactor the site vertical API.

At that point we'll have a better idea about how we're going to use this parameter, more specifically, whether `segment_id={1,3,4,5...}` might be more suitable (?)

I don't know for sure, just flagging the idea.

Using a slug is also fine, and less abstract than using numbers as ids, though it seems more susceptible to case/spelling errors.

Reply:
@michaeldcain has spun up a patch to accept `site_type` as a parameter {Integer}

D28623-code

Maybe later we can unify the vars to refer to `segment` or `site_segment` to reduce confusion. A nice janitorial task 🔨

Reply:
I don't think we'll need to pass many segments IDs for the time being, see https://github.com/Automattic/zelda-private/issues/1#issuecomment-495149632.

Good note on using the ID instead of slug, I'll update this PR.

Reply:
I'm personally not a fan of keying by ID's that are integers - it could cause a little confusion down the line as it looks like we're accessing values in an array, and '1' doesn't give us the same context as 'business', for example. 
Does it benefit us to do so?
If the only concern is how we receive data from the API then something like this could work for us

```js
const segmentSlugsByID = {
  '1': 'business',
  '2': 'professional'
  // etc.
}

const segmentSlug = segmentSlugsByID[ data.segmentId ];
```

Reply:
> I'm personally not a fan of keying by ID's that are integers - it could cause a little confusion down the line

@spen [This file](https://github.com/Automattic/wp-calypso/blob/356caee85ab18cc4c74c66ae50beb01bab660515/client/lib/signup/site-type.js) acts as a type of dictionary so we can store/access all that guff in one place.

It's bulged a bit around the mid-section in the meantime, but the plan is to deliver all the relevant slug/id information via the segments API so we never have to store it again.

The copy associated with each segment however might still need to live in Calypso, so there'd be some merge operation or lookup based on the segment API response.

Reply:
Ignore my last comment, I got the wrong end of the stick.

You're referring to the `state.signup.verticals[key]`, right?

My original comment was referring strictly to the the value we send to the verticals API, which I think should be the integer, not the state array keys.

In that case, I tend to agree with @spen

Reply:
Ahh yes sorry, that's exactly what I mean!
It might have made a bit more sense to have started a new convo on one of those more direct lines :)